src/cli_config.rs

```diff
@@ -0,0 +1,92 @@
+use std::net::ToSocketAddrs;
+use std::num::ParseIntError;
+use std::str::FromStr;
+
+use clap::{Arg, App, AppSettings};
+use hex::FromHex;
+use itertools::Itertools;
+use tox::toxcore::crypto_core::*;
+use tox::toxcore::dht::packed_node::PackedNode;
+
+/// Config for threading.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum ThreadsConfig {
+    /// Detect number of threads automatically by the number of CPU cores.
+    Auto,
+    /// Exact number of threads.
+    N(u16)
+}
+
+impl FromStr for ThreadsConfig {
+    type Err = ParseIntError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s == "auto" {
+            Ok(ThreadsConfig::Auto)
+        } else {
+            u16::from_str(s).map(ThreadsConfig::N)
+        }
+    }
+}
+
+/// Config parsed from command line arguments.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct CliConfig {
+    /// List of bootstrap nodes.
+    pub bootstrap_nodes: Vec<PackedNode>,
+    /// Number of threads for execution.
+    pub threads_config: ThreadsConfig,
+}
+
+/// Parse command line arguments.
+pub fn cli_parse() -> CliConfig {
+    let matches = App::new(crate_name!())
+        .version(crate_version!())
+        .author(crate_authors!("\n"))
+        .about(crate_description!())
+        .setting(AppSettings::ColoredHelp)
+        .arg(Arg::with_name("bootstrap-node")
+            .short("b")
+            .long("bootstrap-node")
+            .help("Node to perform initial bootstrap")
```

Comment:
How about giving examples of arguments.
ex) "1.1.1.2:33445", "a0b0c01122334455667788bbaacfdd....."

Reply:
`--help` gives us: `-b, --bootstrap-node <public key> <address>    Node to perform initial bootstrap`, which is fine